src/peerstate.rs

```diff
@@ -216,9 +218,10 @@ impl Peerstate {
                      FROM acpeerstates  \
                      WHERE verified_key_fingerprint=? \
                      OR addr=? COLLATE NOCASE \
-                     ORDER BY verified_key_fingerprint=? DESC, last_seen DESC LIMIT 1;";
+                     ORDER BY verified_key_fingerprint=? DESC, addr=? COLLATE NOCASE DESC, \
```

Comment:
Note that this is one of key changes here -- if AEAP didn't happen before for some good reason, we must return the peerstate corresponding to the peer's address, not  just the recent one.